open_humans/views.py

```diff
@@ -283,15 +280,36 @@ def get_context_data(self, **kwargs):
         return context
 
 
-class DataRetrievalTaskDeleteView(PrivateMixin, DeleteView):
+class SourceDataFilesDeleteView(PrivateMixin, DeleteView):
     """
-    Let the user delete a dataset.
+    Let the user delete all datafiles for a source.
     """
+    template_name = 'member/my-member-source-data-files-delete.html'
     success_url = reverse_lazy('my-member-research-data')
 
     def get_queryset(self):
-        return DataRetrievalTask.objects.filter(user=self.request.user)
+        source = self.kwargs['source']
+
+        data_file_model = app_name_to_data_file_model(source)
+
+        return data_file_model.objects.filter(
+            user_data__user=self.request.user)
+
+    def get_object(self, queryset=None):
```

Comment:
But as a result, self.object is not actually an object, it's a queryset. It works because calling delete() on a queryset also works? But it seems like the language of DeleteView anticipates an object: https://github.com/django/django/blob/master/django/views/generic/edit.py#L257